oauth2/src/main/java/com/okta/spring/oauth/OktaOAuthProperties.java

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright 2017 Okta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.okta.spring.oauth;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+import javax.annotation.PostConstruct;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@ConfigurationProperties("okta")
+public class OktaOAuthProperties {
```

Comment:
I'm a little confused on the difference between `OkatOAuthProperties` and `OAuthProperties` and why `OAuthProperties` is embedded in `OktaOAuthProperties`.

Reply:
This brings up a few questions, i'm not sure the _best_ way to handle the properties object.
But your point is valid, the top level should be renamed to `OktaProperties` (on it)

The odd nested `OAuthProperties getOauth2()`/`oauth2` to make Spring's `ConfigurationProperties` work out of the box.

it might be worth breaking this up into multiple classes, one for each `okta.*` so one for `okta.client`, `okta.oauth2` and another for `okta.web` (this last one is already separate)

The `okta.client` properties are defined by the the SDK's spec so they are reused where possible